### PR TITLE
Add support for high bit depth multichannel images

### LINKED ISSUE #1888: Tracking Issue for high bit depth multichannel images
Pillow (and PIL) is currently able to open 8 bit per channel multi-channel images (such as RGB) but is able to open higher bit depth images (e.g. I16, I32, or Float32 images) if they are single channel (e.g., grayscale). 

## Previous References

This has been requested many times: #1828, #1885, #1839, #1602, and farther back.

## Requirements

- We should be able to support common GIS formats as well as high bit depth RGB(A) images. 
- At least 4 channels, but potentially more (see #1839)
- Different pixel formats, including I16, I32, and Float.

## Background Reference Info

The rough sequence for image loading is:

- Image file is opened
- Each of the ImagePlugin _accept functions have a chance to look at the first few bytes to determine if they should attempt to open the file
- The `*ImagePlugin._open` method is called giving the image plugin a chance to read more of the image and determine if it still wants to consider it a valid image of it's particular type. If it does, it passes back a tile definition which includes a decoder and an image size. 
- If there is a successful _open call, at some point later `*ImagePlugin._load` may be called on the image, which runs the decoder producing a set of bytes in a raw mode. This is where things like compression are handled, but the output of the decoder is not necessarily what we're storing in our internal structures. 
- The image is unpacked (`Unpack.c`) from the raw mode (e.g. `I16;BS`) into a storage (`Storage.c`) mode (`I`).
- It's now possible to operate on the image (e.g. crop, pixel access, etc)

 There are 3 (or 4) image data pointers, as defined in Imaging.h:
```
struct ImagingMemoryInstance {

    /* Format */
    char mode[IMAGING_MODE_LENGTH];	/* Band names ("1", "L", "P", "RGB", "RGBA", "CMYK", "YCbCr", "BGR;xy") */
    int type;		/* Data type (IMAGING_TYPE_*) */
    int depth;		/* Depth (ignored in this version) */
    int bands;		/* Number of bands (1, 2, 3, or 4) */
    int xsize;		/* Image dimension. */
    int ysize;

    /* Colour palette (for "P" images only) */
    ImagingPalette palette;

    /* Data pointers */
    UINT8 **image8;	/* Set for 8-bit images (pixelsize=1). */
    INT32 **image32;	/* Set for 32-bit images (pixelsize=4). */

    /* Internals */
    char **image;	/* Actual raster data. */
    char *block;	/* Set if data is allocated in a single block. */

    int pixelsize;	/* Size of a pixel, in bytes (1, 2 or 4) */
    int linesize;	/* Size of a line, in bytes (xsize * pixelsize) */

    /* Virtual methods */
    void (*destroy)(Imaging im);
};

```

The only one that is guaranteed to be set is `**image`, which is an array of pointers to row data. 

## Changes Required

[to be continued]

### PR DESCRIPTION
**Pull request by aclark4life**
Fixes #1888 .

Changes proposed in this pull request:

 * Add support for high bit depth multichannel images

### CODE REVIEW

**Comment by hugovk** [comment]
Please can you rename the PR to something more descriptive?

File: PIL/TiffImagePlugin.py
Comment (hugovk) [inline PIL/TiffImagePlugin.py:1]: Is this an old file added by accident? 

The source is now under src/PIL/ not PIL/, and this file has handling for Python 2.6.

**Comment by yoursunny** [comment]
My contribution from #8161 has been reverted 🫥

**Comment by hugovk** [comment]
@yoursunny Would you prefer to open this as a PR yourself directly?

**Comment by aclark4life** [comment]
@yoursunny @radarhere @hugovk OK sorry about that, I see that we can grab that contribution from [here](https://github.com/python-pillow/Pillow/tree/6fb6106a8fa467a107540aad8d36a040a1826aad) so let's try again with my apologies for the extra work. I'm going to close this PR, create a new PR from @yoursunny 's [hack branch](https://github.com/python-pillow/Pillow/tree/hack202406) and regroup there. I'm also going to remove the branch I created. Maybe we can commit directly to the PR in the future. 🤔

**Pull request closed without merging**